src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -16,15 +28,26 @@ export const TodoFilter = () => (
         type="text"
         className="input"
         placeholder="Search..."
+        onChange={hadleSearchInputChange}
+        value={searchQuery}
       />
       <span className="icon is-left">
         <i className="fas fa-magnifying-glass" />
       </span>
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
-      </span>
+      {searchQuery.length > 0 && (
+        <span className="icon is-right" style={{ pointerEvents: 'all' }}>
+          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+          <button
+            data-cy="clearSearchButton"
+            type="button"
+            className="delete"
+            onClick={() => {
+              setSearchQuery('');
+            }}
```

Comment:
```suggestion
            onClick={() => setSearchQuery('')}
```

Reply:
For me, this correction happens automatically. I thought it was the Prettier setting I got with the project.